compiler/rustc_trait_selection/src/traits/select/confirmation.rs

```diff
@@ -340,7 +340,7 @@ impl<'cx, 'tcx> SelectionContext<'cx, 'tcx> {
         let predicate =
             self.tcx().erase_regions(self.tcx().erase_late_bound_regions(obligation.predicate));
 
-        let Some(assume) = rustc_transmute::Assume::from_const(
+        let Some(Ok(assume)) = rustc_transmute::Assume::from_const(
```

Comment:
for `None` you have to return ambiguity, not `Err(Unimplemented)` :thinking:

Reply:
Is there an existing way to represent ambiguity in the current solver? Just return an empty list of candidates? Was going through the code in the `select` directory, but it's not clear to me